spdx/v2/v2_2/package.go

```diff
@@ -122,6 +123,20 @@ type Package struct {
 	hasFiles []common.DocElementID
 }
 
+var emptyVerificationCodeRegex = regexp.MustCompile(`,\s*"packageVerificationCode"\s*:\s*\{\s*"packageVerificationCodeValue"\s*:\s*""\s*}`)
+
+func (p *Package) MarshalJSON() ([]byte, error) {
+	type pkg Package
+	p2 := pkg(*p)
+
+	data, err := json.Marshal(p2)
+	if err != nil {
+		return nil, err
+	}
+
+	return emptyVerificationCodeRegex.ReplaceAll(data, []byte("")), nil
```

Comment:
hmm, according to the 2.2 spec, even if this was not empty, it has to be omitted no matter what the value is. In this case, it seems like we would have to check the value and provide an error if it is non-empty as long as `filesAnalyzed=false`?

(on the side, i have a preference from unmarshalling into a map and then into the struct)

Reply:
I updated this with a better decode/delete property/re-encode to remove the property (instead of using a regex, as it was before). I don't really want to alter the behavior much more than this, however. We have a lot of instances that we output whatever data the struct has rather than attempting to always output something according to the spec, so would probably rather not start here but would definitely be open to this in the future, especially once we work on SPDX 3 support.